.github/workflows/build.yaml

```diff
@@ -1,83 +1,151 @@
+    steps:
+      -
+        name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      -
+        name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+      -
+        name: Test
+        uses: docker/bake-action@v1
+        with:
+          targets: test
+      -
+        name: Upload coverage
+        uses: codecov/codecov-action@v2
+        with:
+          file: ./coverage.txt
+
+  test-host:
     strategy:
+      fail-fast: false
       matrix:
-        os: [ubuntu-20.04, macos-latest]
+        os:
+          - macos-latest
     runs-on: ${{ matrix.os }}
     steps:
-      - uses: actions/checkout@v2
-      - name: Setup Go
+      -
+        name: Checkout
+        uses: actions/checkout@v2
+      -
+        name: Setup Go
         uses: actions/setup-go@v2
         with:
           go-version: ^1.15
-      - name: Install Linux dependencies
-        if: matrix.os == 'ubuntu-20.04'
-        run: sudo apt-get install -y libjq-dev libonig-dev gcc
-      - name: Build executable
-        run: make build
+      -
+        name: Test
+        run: make test
 
-  docker:
-    name: Build Docker image
-    runs-on: ubuntu-20.04
+  build:
+    runs-on: ubuntu-latest
+    needs:
+      - validate
+      - test-ctn
+      - test-host
     steps:
-      - name: Checkout
+      -
+        name: Checkout
         uses: actions/checkout@v2
-      - name: Set outputs
-        id: vars
-        run: echo "::set-output name=ref::$(git status | head -1 | awk '{print $NF}' | sed 's#/#_#g')"
-      - name: Set up QEMU
-        uses: docker/setup-qemu-action@v1
-      - name: Set up Docker Buildx
-        uses: docker/setup-buildx-action@v1
         with:
-          driver-opts: image=moby/buildkit:master
-      - name: Build
-        uses: docker/build-push-action@v2
+          fetch-depth: 0
+      -
+        name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v3
         with:
-          push: false
+          images: |
+            ${{ env.GHCR_SLUG }}
           tags: |
-            quay.io/jzelinskie/faq:${{ steps.vars.outputs.ref }}
-      - name: Login to Quay
+            type=ref,event=tag
+            type=ref,event=pr
+            type=edge
+          flavor: |
+            latest=${{ startsWith(github.ref, 'refs/tags/') }}
+          labels: |
+            org.opencontainers.image.title=faq
+            org.opencontainers.image.description=Format Agnostic jQ -- process various formats with libjq
+      -
+        name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+      -
+        name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+      -
+        name: Login to GHCR
+        if: github.event_name != 'pull_request'
         uses: docker/login-action@v1
-        if: github.event_name == 'push'
         with:
-          registry: quay.io
-          username: jzelinskie+faq_gha
-          password: ${{ secrets.QUAY_ROBOT_TOKEN }}
```

Comment:
Oh btw, do you still want to push on Quay? Last time I checked, this registry didn't support multi-platform images.

Reply:
Quay does support multi-platform images. I looked and there is regular traffic pulling from there, so ideally we don't break those users.

Reply:
Ok will bring back Quay!